ui/propertyeditor/propertytexteditor.cpp

```diff
@@ -52,39 +52,42 @@ QString PropertyTextEditorDialog::text() const
 }
 
 PropertyTextEditor::PropertyTextEditor(QWidget *parent)
-    : QLineEdit(parent)
+    : PropertyExtendedEditor(parent)
+    , m_lineEdit(new QLineEdit)
 {
-    auto *hl = new QHBoxLayout(this);
-    hl->setMargin(0);
-    hl->setSpacing(1);
-
-    auto *editButton = new QToolButton(this);
-    editButton->setText(QStringLiteral("..."));
-
-    QMargins margins(textMargins());
-    margins.setRight(editButton->sizeHint().width() + hl->spacing());
-
-    setFrame(false);
-    setTextMargins(margins);
+    setInlineWidget(m_lineEdit, [](QWidget *widget, const QString &text){
+        auto edit = qobject_cast<QLineEdit*>(widget);
+        edit->setText(text);
+    });
+    connect(m_lineEdit, SIGNAL(editingFinished()), this, SLOT(editFinished()));
+}
 
-    hl->addStretch();
-    hl->addWidget(editButton);
+void PropertyTextEditor::edit()
+{
+    PropertyTextEditorDialog dlg(value().toString(), this);
+    if (dlg.exec() == QDialog::Accepted)
+        save(dlg.text());
+}
 
-    connect(editButton, SIGNAL(clicked()), SLOT(edit()));
+void PropertyTextEditor::editFinished()
+{
+    save(m_lineEdit->text());
 }
 
-void PropertyTextEditor::save(const QString &text)
+PropertyByteArrayEditor::PropertyByteArrayEditor(QWidget *parent)
+    : PropertyTextEditor(parent)
 {
-    setText(text);
 
-    // The user already pressed Apply, don't force her/him to do again
-    QKeyEvent event(QEvent::KeyPress, Qt::Key_Enter, Qt::NoModifier);
-    QApplication::sendEvent(this, &event);
 }
 
-void PropertyTextEditor::edit()
+void PropertyByteArrayEditor::edit()
 {
-    PropertyTextEditorDialog dlg(text(), this);
+    PropertyTextEditorDialog dlg(value().toByteArray(), this);
```

Comment:
Let be clear we assume all utf8 by default and for Qt4 support, let use QString::fromUtf8()